src/util/system.h

```diff
@@ -367,8 +367,8 @@ std::string CopyrightHolders(const std::string& strPrefix);
  * On platforms that support it, tell the kernel the calling thread is
  * CPU-intensive and non-interactive. See SCHED_BATCH in sched(7) for details.
  *
- * @return The return value of sched_setschedule(), or 1 on systems without
- * sched_setschedule().
+ * @return The return value of pthread_setschedparam(), or 1 on systems without
+ * SCHED_BATCH.
```

Comment:
I don't think it's necessary to describe implementation details here at all. Maybe just state it returns 0 on success and any other value on failure or when the functionality is not supported on the OS.

Note that we don't use the return value at all, not even for logging.

I think I'd prefer making this a purely advisory function, that returns `void`.

Reply:
> I think I'd prefer making this a purely advisory function, that returns void.

I've updated the PR to do that.